R/getGenomicTiles.R

```diff
@@ -0,0 +1,180 @@
+#' @title Get regions tiling a genome.
+#'
+#' @description Get sequential, potentially annotated regions of a fixed lengths
+#'    (tiles) along chromosomes of a genome.
+#'
+#' @author Michael Stadler
+#'
+#' @param genome The genome to work on. Either a \code{\link[BSgenome]{BSgenome}}
+#'     object, a \code{character} scalar with the name of an installed \code{\link[BSgenome]{BSgenome}}
+#'     or with a file path and name pointing to a fasta file with the genome sequence,
+#'     or a named \code{numeric} vector giving the names and lengths of chromosomes.
+#' @param tileWidth \code{numeric} scalar with the tile length.
+#' @param hasOverlap Named \code{list} with \code{\link[GenomicRanges]{GRanges}} object(s).
```

Comment:
I don't think this is strictly necessary, but I think we could probably also support `GRangesList` objects (as elements of the named list) without too many changes downstream - I'm thinking of e.g. transcripts, or other non-consecutive sets of regions that still have the same identifier (and thus should be counted as one overlapping range).

Reply:
Yes, this could be done, although I can currently not think of a use-case. How would you define an overlap (with the region spanned by the ranges, or with a range)? Could it also be achieved by unlisting the `GRangesList` (and repeating the name for all of the ranges in an element)?

Reply:
More specifically, given `grl <- GRangesList(e1 = GRanges("chr1", IRanges(c(1,11), width = 3)))`, should we downstream use

- `unlist(grl)` or
- `unlist(range(grl))`

Reply:
So I would use `unlist(grl)` - I think that if you want the `range()` you should provide those ranges in the first place. I guess my only issue with unlisting is that if you then count the number of overlapping ranges you would count the ranges corresponding to the same entry in the `GRangesList` separately (and thus one `GRangesList` entry, e.g., one gene or transcript, can contribute >1 count, which may be counterintuitive) - to avoid that we'd have to count the number of unique range identifiers.

Reply:
I added `unlist(grl)` to "fracOverlap" and "nearest", but not to "numOverlap" (or "hasOverlap"), where it would have caused the problem you mention (or was not needed in the first place). This seems to do what is expected.